src/fftr.nim

```diff
@@ -323,25 +323,105 @@ func process*(
   for k in 0..<result.len:
     for n in 0..<result.len:
       result[k] += input[n] * twiddles[k * n mod twiddles.len]
 
-func fft*(input: openArray[Complex64], inverse: bool): seq[Complex64] =
-  # Convenience entry point that chooses FFT based on input length
-  result.setLen(input.len)
-
-  if isPowerOfTwo(input.len):
-    var ctx = Stockham2.init(input.len, inverse)
+  if inverse and normalize:
+    let complexInputLen = complex64(float64(input.len))
+    for i in 0..<result.len:
+      result[i] = result[i] / complexInputLen
+
+func dft*(input: openArray[Complex64], inverse: bool = false, normalize: bool = false, n = 0): seq[Complex64] =
+  ## Calculates the DFT or the IDFT of an input signal using a slow, straight-forward implementation
+  ##
+  ## This function is only meant for testing, since it is a straight-forward but slow implementaiton of the DFT
+  ##
+  ## Inputs:
+  ## - input: The input signal.
+  ## - inverse: if true, calculates the inverse FFT, otherwise calculates the FFT (which is the default).
+  ## - normalize: if true, normalizes the results by dividing by the length of the signal when inverse is true.
+  ## - n: Length of the the output.
+  ##      If `n` is not given, the length of the input is automatically used.
+  ##      If `n` is smaller than the length of the input, the input is cropped.
+  ##      If `n` is larger, the input is padded with zeros.
+  let size = if n <= 0: input.len else: n
+
+  if size < input.len:
+    # Extend the input with zeros as much as needed
+    let input = input.toSeq & newSeq[Complex64](size - input.len)
+  elif size < input.len:
+    # Crop the input as much as needed
+    let input = input[0..<size]
+
+  dftImpl(input, inverse=inverse, normalize=normalize)
+
+func fftImpl(input: openArray[Complex64], inverse: bool = false, normalize: bool = false): seq[Complex64] =
+  ## Calculates the FFT or the IFFT of an input signal using the best method given the input length
+  ##
+  ## Inputs:
+  ## - input: The input signal
+  ## - inverse: if true, calculates the inverse FFT, otherwise calculates the FFT
+  ## - normalize: if true, normalizes the results by dividing by the length of the signal when inverse is true
+  let size = input.len
+  result.setLen(size)
+
+  if isPowerOfTwo(size):
+    var ctx = Stockham2.init(size, inverse)
     process(ctx, input, result)
-  elif input.len.countTrailingZeroBits == 0:
-    var ctx = Bluestein.init(input.len, inverse)
+  elif size.countTrailingZeroBits == 0:
+    var ctx = Bluestein.init(size, inverse)
     process(ctx, input, result)
   else:
-    var ctx = RadixMixed.init(input.len, inverse)
+    var ctx = RadixMixed.init(size, inverse)
     process(ctx, input, result)
+
+  if inverse and normalize:
+    let complexInputLen = complex64(float64(size))
+    for i in 0..<result.len:
+      result[i] = result[i] / complexInputLen
+
+func fft*(input: openArray[Complex64], inverse: bool = false, normalize: bool = false, n = 0): seq[Complex64] =
```

Comment:
I knew that scipy's fft had 3 normalization options. If you think that makes sense I can support them all by passing a norm argument that would be an enum.

Regarding n, I think it would be a good idea to support it as well, following scpy's API as much as possible.

Reply:
I've added `norm` arguments matching SCIPY's options plus a "disabled" option to completely disable the normalization. I've made the default `backwards` to mach SCIPY.